src/EnergyPlus/UnitarySystem.cc

```diff
@@ -17856,5 +17891,147 @@ namespace UnitarySystems {
+                        // determine this speed's runtime
+                        highSpeedEconRuntime = (std::abs(zoneLoad) - this->m_LowSpeedEconOutput) / (econClgOutput - this->m_LowSpeedEconOutput);
+                    }
+                } else {
+                    highSpeedEconRuntime = 1.0;
+                }
+                // set economizer air flow "speed"
+                this->m_EconoSpeedNum = clgSpd;
+                // set economizer PLR, a.k.a the system fan part load ratio, and runtime at each speed
+                this->m_EconoPartLoadRatio = highSpeedEconRuntime;
+                this->m_LowSpeedEconRuntime = 1 - highSpeedEconRuntime;
+                break;
+            }
+        }
+    }
+
+    void UnitarySys::calcMixedTempAirSPforEconomizerStagingOperation(EnergyPlusData &state,
+                                                                     int const airLoopNum,
+                                                                     bool const firstHVACIteration,
+                                                                     Real64 const zoneLoad)
+    {
+        auto outdoorAirController = state.dataMixedAir->OAController(this->OAControllerIndex);
+        using Psychrometrics::PsyCpAirFnW;
+        Real64 cpAir = PsyCpAirFnW(state.dataLoopNodes->Node(outdoorAirController.InletNode).HumRat);
+        Real64 zoneTemp = state.dataLoopNodes->Node(state.dataZoneEquip->ZoneEquipConfig(this->ControlZoneNum).ZoneNode).Temp;
+        // determine and set new air loop mixed air flow rate
+        Real64 mixedAirFlowRate = this->m_EconoPartLoadRatio * this->m_CoolMassFlowRate[this->m_EconoSpeedNum] +
+                                  this->m_LowSpeedEconRuntime * this->m_CoolMassFlowRate[this->m_EconoSpeedNum - 1];
+        if (airLoopNum > 0) {
+            // request fixed mixed flow rate
+            state.dataAirLoop->AirLoopControlInfo(airLoopNum).LoopFlowRateSet = true;
+            state.dataAirLoop->AirLoopFlow(airLoopNum).ReqSupplyFrac = mixedAirFlowRate / this->m_CoolMassFlowRate[this->m_NumOfSpeedCooling];
+            // adjust mixed air flow rate for rated air flow rate adjustment (variable speed coils only)
+            state.dataAirLoop->AirLoopFlow(airLoopNum).ReqSupplyFrac *=
+                this->m_CoolMassFlowRate[this->m_NumOfSpeedCooling] / state.dataAirLoop->AirLoopFlow(airLoopNum).DesSupply;
+        }
+        // determine air temperature difference across fan based on new mixed air flow rate
+        int mixedAirNode = outdoorAirController.MixNode;
+        Real64 fanDTAtMixedAirFlowRate = this->getFanDeltaTemp(
+            state, firstHVACIteration, mixedAirFlowRate, mixedAirFlowRate / this->m_CoolMassFlowRate[this->m_NumOfSpeedCooling]);
+        // determine new mixed air setpoint
+        Real64 newMixedAirSP = zoneTemp - std::abs(zoneLoad) / (cpAir * mixedAirFlowRate);
+        state.dataLoopNodes->Node(mixedAirNode).TempSetPoint = newMixedAirSP - fanDTAtMixedAirFlowRate;
```

Comment:
@lymereJ this new control puts a SP on the mixed air node that tries to control the OA system to meet the zone load. That's fine, but 1) is stomping on the mixed air SPM, and 2) the user does not know anything about this override. In fact I'm not even sure this is doing anything because the SPMs and OA system are simulated before the UnitarySystem. Here's another case where I want to know what happens if this SP control is removed. If it's necessary then fine, but the user needs to know what is happening in their simulation. If this in fact is needed then I think this should be described in the docs and the "Unitary System Mixed Air Setpoint Temperature" be reported out to the user.